Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/CoinsConfig.hs

```diff
@@ -0,0 +1,113 @@
+import qualified Kernel.Types.Beckn.Context
+import qualified Kernel.Types.Id as ID
+import qualified Kernel.Utils.Common as UC
+import Servant
+import qualified Storage.CachedQueries.CoinsConfig as CQConfig
+import qualified Storage.Queries.Coins.CoinsConfig as QConfig
+import Storage.Queries.Translations (create)
+import Storage.Queries.TranslationsExtra (isTranslationExist)
+import Tools.Auth
+
+putCoinsConfigUpdate ::
+  ID.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  Common.UpdateReq ->
+  Environment.Flow Kernel.Types.APISuccess.APISuccess
+putCoinsConfigUpdate _merchantShortId _opCity Common.UpdateReq {..} = do
+  mbCoinsConfig <- QConfig.fetchConfigOnIdBasis $ ID.cast entriesId
+  case mbCoinsConfig of
+    Nothing -> error "No entries with such Id:" entriesId
+    Just coinsConfig -> do
+      let updatedCoinsConfig =
+            coinsConfig
+              { DTCC.active = active,
+                DTCC.expirationAt = expirationAt,
+                DTCC.coins = coins
+              }
+          vehicleCategory = getVehicleCategoryFromMaybe coinsConfig.vehicleCategory
+      _ <- QConfig.updateCoinEntries updatedCoinsConfig
+      _ <- CQConfig.clearCache coinsConfig.eventName coinsConfig.eventFunction (ID.Id coinsConfig.merchantOptCityId) vehicleCategory
+      pure Success
+
+postCoinsConfigCreate ::
+  ID.ShortId Domain.Types.Merchant.Merchant ->
+  Kernel.Types.Beckn.Context.City ->
+  Common.CreateCoinsConfigReq ->
+  Environment.Flow Kernel.Types.APISuccess.APISuccess
+postCoinsConfigCreate _merchantShortId _opCity req = do
+  uuid <- UC.generateGUIDText
+  (newCoinsConfig, eventMessageLs) <- case req of
+    Common.NewCoinsConfig (Common.NewCoinsConfigReq {..}) ->
+      pure
+        ( DTCC.CoinsConfig
+            { id = ID.Id uuid,
+              ..
+            },
+          eventMessages
+        )
+    Common.DuplicateCoinsConfig (Common.DuplicateCoinsConfigsReq {..}) -> do
+      mbCoinsConfig <- QConfig.fetchConfigOnIdBasis $ ID.cast entriesId
+      case mbCoinsConfig of
+        Nothing -> error "No entries with such Id:" entriesId
+        Just coinsConfig ->
+          pure
+            ( coinsConfig
+                { DTCC.id = ID.Id uuid,
+                  DTCC.eventFunction = eventFunction
+                },
+              eventMessages
+            )
+  _ <- do
+    QConfig.updateCoinEntries newCoinsConfig
+    CQConfig.clearCache
+      newCoinsConfig.eventName
+      newCoinsConfig.eventFunction
+      (ID.Id newCoinsConfig.merchantOptCityId)
+      (getVehicleCategoryFromMaybe newCoinsConfig.vehicleCategory)
+    processingTranslations newCoinsConfig eventMessageLs
+  pure Success
+
+getVehicleCategoryFromMaybe :: Maybe VehicleCategory -> VehicleCategory
+getVehicleCategoryFromMaybe mbVehicleCategory =
+  case mbVehicleCategory of
+    Just vehicleCategory -> vehicleCategory
+    Nothing -> error "vehicleCategory is Nothing" mbVehicleCategory
```

Comment:
the same, remove `error` function
do we really need to throw error here, if `vehicleCategory` used only for clear cache?